test/qa_node.cpp

```diff
@@ -81,16 +111,43 @@ interpolation_decimation_test(const TestData &data, std::shared_ptr<fair::thread
     int_dec_block.denominator = data.denominator;
-    if (data.out_port_max_samples >= 0) int_dec_block.out.max_samples = static_cast<int>(data.out_port_max_samples);
-    if (data.out_port_min_samples >= 0) int_dec_block.out.min_samples = static_cast<int>(data.out_port_min_samples);
+    if (data.out_port_max >= 0) int_dec_block.out.max_samples = static_cast<int>(data.out_port_max);
+    if (data.out_port_min >= 0) int_dec_block.out.min_samples = static_cast<int>(data.out_port_min);
 
     std::ignore = flow.connect<"out">(source).to<"in">(int_dec_block);
     auto sched  = scheduler(std::move(flow), thread_pool);
     sched.run_and_wait();
 
-    expect(eq(int_dec_block.status.process_counter, data.expected_counter)) << "process_bulk invokes counter";
-    expect(eq(int_dec_block.status.n_inputs, data.expected_in)) << "number of input samples";
-    expect(eq(int_dec_block.status.n_outputs, data.expected_out)) << "number of output samples";
+    expect(eq(int_dec_block.status.process_counter, data.exp_counter)) << "process_bulk invokes counter, parameters = " << data.to_string();
+    expect(eq(int_dec_block.status.n_inputs, data.exp_in)) << "last number of input samples, parameters = " << data.to_string();
+    expect(eq(int_dec_block.status.n_outputs, data.exp_out)) << "last number of output samples, parameters = " << data.to_string();
+}
+
+void
+stride_test(const StrideTestData &data, std::shared_ptr<fair::thread_pool::BasicThreadPool> thread_pool) {
+    using namespace boost::ut;
+    using scheduler = fair::graph::scheduler::simple<>;
+
+    fg::graph flow;
+    auto     &source          = flow.make_node<CountSource<int>>();
+    source.n_samples          = static_cast<int>(data.n_samples);
+
+    auto &int_dec_block       = flow.make_node<IntDecBlock<int>>();
+    int_dec_block.numerator   = data.numerator;
+    int_dec_block.denominator = data.denominator;
+    int_dec_block.stride      = data.stride;
+    if (data.in_port_max >= 0) int_dec_block.in.max_samples = static_cast<int>(data.in_port_max);
+    if (data.in_port_min >= 0) int_dec_block.in.min_samples = static_cast<int>(data.in_port_min);
+
+    std::ignore = flow.connect<"out">(source).to<"in">(int_dec_block);
+    auto sched  = scheduler(std::move(flow), thread_pool);
+    sched.run_and_wait();
+
+    expect(eq(int_dec_block.status.process_counter, data.exp_counter)) << "process_bulk invokes counter, parameters = " << data.to_string();
+    expect(eq(int_dec_block.status.n_inputs, data.exp_in)) << "last number of input samples, parameters = " << data.to_string();
+    expect(eq(int_dec_block.status.n_outputs, data.exp_out)) << "last number of output samples, parameters = " << data.to_string();
+    expect(eq(int_dec_block.status.total_in, data.exp_total_in)) << "total number of input samples, parameters = " << data.to_string();
+    expect(eq(int_dec_block.status.total_out, data.exp_total_out)) << "total number of output samples, parameters = " << data.to_string();
```

Comment:
Counting the effectively generated number of samples is good but in case of the stride it might IMO be pertinent to also check for a sequence type structure, i.e. having `stride:=3` and `N_IN_PORT_MIN_MAX:=5` would -- for example -- extract for the sequence

{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, ...}

the following output:

{1, 2, 3, 4, 5, /*skip 3 samples*/ 9, 10, 11, 12, 13, /*skip 3 samples*/ 17, ...}

Reply:
I think we discussed last time that samples should start always from the beginning of the stride. But in your example it is not the case. 
The output from you example should be: {1, 2, 3, /skip 2/, 6, 7, 8, /skip 2/, 11, 12, 13, /skip 2/  ...} 
In this case you have always a constant distance between two data processing.

I will also add additional tests.

Reply:
I think we discussed last time that samples should start always from the beginning of the stride. 
But in your example it is not the case. If stride < N means that one has overlapping.

The output from you example should be: {1, 2, 3, 4, 5, | 4, 5, 6, 7, 8, | 7, 8, 9. 10, 11 | ...}

I will also add additional tests.

Reply:
@drslebedev yes, was a typo/error on my side meant `stride:=8` in the example above. I like the new additional tests! :+1: